exercises/go-counting/canonical-data.json

```diff
@@ -0,0 +1,172 @@
+{
+  "exercise": "go-counting",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Test for black territory on 5x5 board",
```

Comment:
Let's have descriptions say **why** a test case is present, rather than **what** it is testing. For example, some new functionality that this case tests that previous cases did not test. For example, "territory surrounded entirely by stones" versus "territory bordering the edge of the board", etc.